### ISSUE TITLE
Error after upgrading from 0.2.7 to 0.2.10

### ISSUE THREAD

**Issue by BolunDai0216** [Simple-Robotics/proxsuite#123]
Hi,

I was running my code [here](https://github.com/BolunDai0216/fr3_ros/blob/main/fr3_ros/src/cbf_controller.cpp). And I noticed that even though I did not change anything, after upgrading `ProxSuite` from `v0.2.7` to `v0.2.10` I get the following error:

```console
terminate called after throwing an instance of 'std::invalid_argument'
  what():  From file: /home/bolun/miniconda3/envs/fr3_env/include/proxsuite/proxqp/dense/wrapper.hpp
in function: void proxsuite::proxqp::dense::QP<T>::init(proxsuite::optional<Eigen::Ref<const Eigen::Matrix<LhsScalar, -1, -1, 1>, 0, typename Eigen::internal::conditional<const Eigen::Matrix<LhsScalar, -1, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type> >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<Scalar, -1, 1> > >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<LhsScalar, -1, -1, 1>, 0, typename Eigen::internal::conditional<const Eigen::Matrix<LhsScalar, -1, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type> >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<Scalar, -1, 1> > >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<LhsScalar, -1, -1, 1>, 0, typename Eigen::internal::conditional<const Eigen::Matrix<LhsScalar, -1, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type> >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<Scalar, -1, 1> > >, proxsuite::optional<Eigen::Ref<const Eigen::Matrix<Scalar, -1, 1> > >, bool, proxsuite::optional<T>, proxsuite::optional<T>, proxsuite::optional<T>) [with T = double; proxsuite::optional<Eigen::Ref<const Eigen::Matrix<LhsScalar, -1, -1, 1>, 0, typename Eigen::internal::conditional<const Eigen::Matrix<LhsScalar, -1, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type> > = std::optional<Eigen::Ref<const Eigen::Matrix<double, -1, -1, 1, -1, -1>, 0, Eigen::OuterStride<> > >; typename Eigen::internal::conditional<const Eigen::Matrix<LhsScalar, -1, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type = Eigen::OuterStride<>; proxsuite::optional<Eigen::Ref<const Eigen::Matrix<Scalar, -1, 1> > > = std::optional<Eigen::Ref<const Eigen::Matrix<double, -1, 1>, 0, Eigen::InnerStride<1> > >; typename Eigen::internal::conditional<const Eigen::Matrix<Scalar, -1, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type = Eigen::InnerStride<1>; proxsuite::optional<T> = std::optional<double>]
at line: 138
wrong argument size: expected 0, got 14
hint: the dimension wrt the primal variable x variable for initializing g is not valid.
```

After downgrading to `v0.2.7` the error goes away. Any idea what might cause this error? Is this a bug in `ProxSuite`?

**Comment by jcarpent** [comment]
@BolunDai0216 I made some change in #125, but I've not have been able to reproduce it, even before the change of #125.
Could you provide a docker image which reproduces your issue?
@fabinsch Do you have in mind which change may lead to this issue?

**Comment by jcarpent** [comment]
From https://github.com/Simple-Robotics/proxsuite/compare/v0.2.7...v0.2.10, I did not see a change which could lead to such an issue. A docker image should help us to fix a bug if any.

**Comment by fabinsch** [comment]
In the dense wrapper we had no change from [v0.2.7...v0.2.10](https://github.com/Simple-Robotics/proxsuite/compare/v0.2.7...v0.2.10). As @jcarpent suggested, a docker image with the same spirit as provided in this issue [here](https://github.com/Simple-Robotics/proxsuite/issues/119) containing the proxsuite version and an example that is showing your problem can help us to debug your problem faster.

**Comment by fabinsch** [comment]
after re-reading the error message
```
wrong argument size: expected 0, got 14
hint: the dimension wrt the primal variable x variable for initializing g is not valid.
```
and checking with the [wrapper code](https://github.com/Simple-Robotics/proxsuite/blob/main/include/proxsuite/proxqp/dense/wrapper.hpp#L140) where this error comes from. It seems that `model.dim`, which is the dimension of the primal variable was set to `0`. but the vector g is of size `14` -> the primal variable should also be of dim `14`.

Can you please provide the code snippet that you are trying to run @BolunDai0216 ?

**Comment by BolunDai0216** [comment]
> after re-reading the error message
> 
> ```
> wrong argument size: expected 0, got 14
> hint: the dimension wrt the primal variable x variable for initializing g is not valid.
> ```
> 
> and checking with the [wrapper code](https://github.com/Simple-Robotics/proxsuite/blob/main/include/proxsuite/proxqp/dense/wrapper.hpp#L140) where this error comes from. It seems that `model.dim`, which is the dimension of the primal variable was set to `0`. but the vector g is of size `14` -> the primal variable should also be of dim `14`.
> 
> Can you please provide the code snippet that you are trying to run @BolunDai0216 ?

You can find the code I was running here: https://github.com/BolunDai0216/fr3_ros/blob/main/fr3_ros/src/cbf_controller.cpp

**Comment by jcarpent** [comment]
@BolunDai0216 Could you provide a docker image to reproduce the bug? The reference to the code you provided does not explain the bug.

**Comment by BolunDai0216** [comment]
I’m not familiar with dockers, but I’ll see how they work.

**Comment by jcarpent** [comment]
Could you check the dimension of the qp problem here https://github.com/BolunDai0216/fr3_ros/blob/72283849fca34b429e93f005a889af1718f9fa24/fr3_ros/src/cbf_controller.cpp#L226
using the line
```cpp
std::court << "dim: " << qp.model.dim << std::endl;
std::court << "n_eq: " << qp.model.n_eq << std::endl;
std::court << "n_in: " << qp.model.n_in << std::endl;
std::court << "qp_g.size(): " << qp_g.size() << std::endl;
```

**Comment by BolunDai0216** [comment]
```console
dim: 14
n_eq: 7
n_in: 0
qp_g size: 14
```

**Comment by BolunDai0216** [comment]
I can record my screen of me reproducing the issue if that helps...

**Comment by fabinsch** [comment]
Thanks for the video already @BolunDai0216. Did you try the version between 0.2.7 and 0.2.10? It could help us know which version change introduced this weird behavior.

**Comment by jcarpent** [comment]
@BolunDai0216 Did you find your bug?

**Comment by BolunDai0216** [comment]
> @BolunDai0216 Did you find your bug?

I tested 0.2.8 and 0.2.9 and it only had a bug for 0.2.10. 

I also noticed that if I create a new `conda` environment and install 0.2.10 it would work without any issues. I still don't know what happened, but it seems like there are some issues with the `conda` environment I created.

Since the problem is probably from my end, I will close this issue. Thanks for the help!
